### DISCUSSION TITLE
Blog Post: Interaction Nets, Combinators, and Calculus

### DISCUSSION THREAD

**Discussion by zicklag** [HigherOrderCO/HVM1#227] [Show and tell]
Hey folks!

I just published a blog post that makes an attempt at explaining some of the foundational concepts of HVM's execution model: [Interaction Nets, Combinators, and Calculus](https://zicklag.github.io/blog/interaction-nets-combinators-calculus/).

I explain the core ideas, and explain how to understand those twisty diagrams. 😉

Hopefully it can help make sense of some of the theory behind this awesome technology! Let me know if you have any feedback!

**Comment by chkoreff** [comment]
I notice that the embedded links in your post take me to 404 pages.  Are you seeing good links on your end?

**Reply by zicklag** [reply]
Just re-deployed it with a fix! Try refreshing the page and see if that fixes it.

**Comment by mgubi** [comment]
Great post. The images are very nice. I was wondering if would be useful to mark the ports with some symbol (a small dot on one of them?) to help visually in the reduction process.

**Reply by zicklag** [reply]
That's a great idea. I'll try that out once I get the chance. Thanks for the suggestion.

**Reply by zicklag** [reply]
I updated the diagrams to use triangles for the nodes and lining up the nodes more strictly, and I think it turned out much more readable now.

**Comment by chkoreff** [comment]
Voila!  C'est bon, merci beaucoup!

**Reply by zicklag** [reply]
Pas de problem!
